ssl/ssl_test.cc

```diff
@@ -10118,5 +10716,1082 @@ TEST(SSLTest, SessionPrint) {
+    // so that it will return success if given an invalid key of the correct
+    // length. Therefore, the call to server_key_share->Accept() will succeed,
+    // but ultimately, the ciphertext (server's public key) will be garbage,
+    // the server and client will end up with different secrets, and the
+    // overall handshake will eventually fail.
+    {
+      bssl::UniquePtr<SSLKeyShare> server_key_share = bssl::SSLKeyShare::Create(t.group_id);
+      bssl::UniquePtr<SSLKeyShare> client_key_share = bssl::SSLKeyShare::Create(t.group_id);
+      ASSERT_TRUE(server_key_share);
+      ASSERT_TRUE(client_key_share);
+      uint8_t server_alert = 0;
+      uint8_t client_alert = 0;
+      Array<uint8_t> server_secret;
+      Array<uint8_t> client_secret;
+      CBB server_out_public_key;
+      CBB client_out_public_key;
+
+      // Start by having the client Offer() its public key
+      EXPECT_TRUE(CBB_init(&client_out_public_key, t.offer_key_share_size));
+      EXPECT_TRUE(client_key_share->Offer(&client_out_public_key));
+
+      // Then invalidate it by negating the bits in the first byte
+      uint8_t *invalid_client_public_key_buf =
+        (uint8_t *)OPENSSL_malloc(t.offer_key_share_size);
+      ASSERT_TRUE(invalid_client_public_key_buf);
+      const uint8_t *client_out_public_key_data = CBB_data(&client_out_public_key);
+      ASSERT_TRUE(client_out_public_key_data);
+      OPENSSL_memcpy(invalid_client_public_key_buf, client_out_public_key_data,
+                     t.offer_key_share_size);
+      invalid_client_public_key_buf[0] = ~invalid_client_public_key_buf[0];
+      Span<const uint8_t> client_public_key =
+        MakeConstSpan(invalid_client_public_key_buf, t.offer_key_share_size);
+
+      // When the server calls Accept() with the invalid public key, it will
+      // return success
+      EXPECT_TRUE(CBB_init(&server_out_public_key, t.accept_key_share_size));
+      EXPECT_TRUE(server_key_share->Accept(&server_out_public_key,
+                                           &server_secret, &server_alert,
+                                           client_public_key));
+
+      // And when the client calls Finish(), it will also return success
+      const uint8_t *server_out_public_key_data = CBB_data(&server_out_public_key);
+      ASSERT_TRUE(server_out_public_key_data);
+      Span<const uint8_t> server_public_key =
+        MakeConstSpan(server_out_public_key_data, CBB_len(&server_out_public_key));
+      EXPECT_TRUE(client_key_share->Finish(&client_secret, &client_alert,
+                                           server_public_key));
+
+      // The shared secrets are of the correct length...
+      EXPECT_EQ(client_secret.size(), t.shared_secret_size);
+      EXPECT_EQ(server_secret.size(), t.shared_secret_size);
+
+      // ... but they are not equal
+      EXPECT_NE(0, OPENSSL_memcmp(client_secret.data(), server_secret.data(),
+                                  t.shared_secret_size));
```

Comment:
Reading these secrets probably isn't valuable but for the future we have a `Bytes` test class which pretty prints any failures and implements comparable. 
```suggestion
      // ... but they are not equal
      EXPECT_NE(Bytes(client_secret), Bytes(server_secret));
```

Reply:
Followed this pattern in all new test code that I have added.